### DISCUSSION TITLE
[FEATURE] Provide custom keys to select elements

### DISCUSSION THREAD

**Discussion by gambhirsharma** [prakhartiwari0/my-web-shortcuts#58] [Ideas]
**Description**
I have been using Neovim for quite a while now. And once you start using vim you can't live with vim-motion. For which I use [Vimium](https://vimium.github.io/) in my browser. One of the best features of it is that it gives you a bunch of defined keys to open elements in the webpage. You don't need to set any key to start using it. 

You press `shift+f` and then some tooltips type element pop and then you can select the element by typing the alphabet. 

### Screenshot. 
<img width="1440" alt="image" src="https://github.com/mywebshortcuts/mywebshortcuts/assets/69895353/5738cdbf-a0c5-43c5-8d6e-9e6b2440d9b8">

**Use Case**
I think this will save a ton of time setting up new key bindings. It will be super useful for new/less visiting websites.

**Comment by 3xp11i** [comment]
I understand that you are saying to have some functionality like Vimium where there are pre-existing shortcuts on websites. But the whole point of this extension is to create your own shortcuts for things you need. 

Can you describe more upon this feature? How do you want to see it, how do you think it's relevant for this tool?

**Comment by 3xp11i** [comment]
@gambhirsharma Please elaborate

**Comment by gambhirsharma** [comment]
I don't know if my suggestion matches your project goals or not. 

But I suggest that you provide some more functionality like the one I mentioned above. I want a key that shows me all the shortcuts given by the User as tooltip-type element to which it will apply, along with this it can also give us some random keybinding for the remaining element for the user which will help navigate them easily. I know you might feel like the second part seems too useless. If so then I suggest you try Vimium for some time and decide on it. I find it super super useful.

If you think you don't need this feature in your extension then it's ok you can close this issue. Thank you :)

**Comment by 3xp11i** [comment]
I won't close this issue but rather convert it into a discussion.

**Comment by 3xp11i** [comment]
My point is that Vimium is targeted at programmers, specifically those who love using Vim. Now a daily user (non-programmer) usually wouldn't like it. I know it depends on person to person, but it's mainly for Vim users. Even I find it cumbersome.


Here are some fundamental differences between Vimium and My Web Shortcuts :point_down: 
> Note: These differences are for the general public, there are always exceptions. 

| Vimium | My Web Shortcuts |
| --- | --- |
| It gives you some predefined shortcuts for "almost" all the elements present in a website | It lets the user create shortcuts for things THEY want and keys THEY want to use. |
| The UI is not user-friendly. At least not for a regular user. | The UI is not just user-friendly, but also attractive (maybe a not-so-important point but still, user experience matters) |
| The predefined shortcuts can interfere with the website shortcuts and vice versa | The user chooses the key, so they can check if a shortcut already exists with that key or not |
It's targeted at VIM users, so the barrier to entry is high for users who never used VIM | This is made for everyone and has a very intuitive interface |
The amount of unnecessary shortcuts it generates for all kinds of elements makes it cumbersome and annoying, there are multiple shortcuts for the same thing | User creates the shortcut they want, so there are no unnecessary shortcuts | 
The shortcuts for each element change when the layout changes, which again makes it a bit irritating | The shortcut remains the same for the element as the user has created it for that particular element | 



So in summary, **Vimium** is for _technical_, _keyboard loving_, _VIM-loving_, people while **My Web Shortcuts** is for anyone who**browses the web*, *multitasks*, and *wants to have some quick key shortcuts* to access stuff in a website. 
The audience for both tools is different, but **My Web Shortcuts can be used by programmers/technical folks too** while **Vimium can't be used by daily non-technical/programmer users.**

**Reply by 3xp11i** [reply]
Also, please understand that these issues with Vimium are for daily users, specifically new users. 
One can get used to anything if given time, but the question is **will a daily user invest their time** in learning all this functionality, reading the docs to understand better, and configuring this tool for their needs, even if their life will become easier with that, **will they do it?**

**Reply by timh-86** [reply]
I am the "non-developer user" that you are talking about... 😛 

I just tried out Vimium and quickly abandoned it because of the complexity, and also because I hardly see a use case. 

Here I am now, happily using My Web Shortcuts. 

I think the developer is right in saying that adding these additional 'Vimium' features will be of little or no interest to My Web Shortcuts users... And more demanding users will just head over to Vimium anyway. 🙂

**Reply by 3xp11i** [reply]
I am very pleased to know this @timh-86! So good to see you're getting value from this little tool. Happy to help you, expect many more awesome features soon ;)

**Comment by gambhirsharma** [comment]
I got the point. I think I am trying to vimify everything without understanding that other users don't like this. 

It is good to see that you understand your audience & build project around their requirements only. Kudos for that. 💯

**Comment by sayande717** [comment]
We could, in the future, have a toggle that turns on `vim mode` for the browser, just like VS Code has extensions for it. It can be a good enhancement.
